docs/aws/audit/apigatewaymonitoring/rules/api_gateway_xray_enabled_remediation.mdx

```diff
@@ -2,6 +2,156 @@
+   aws apigateway create-stage --rest-api-id <rest-api-id> --stage-name <stage-name> --tracing-enabled
+   ```
+
+   Or, if you are updating an existing stage:
+
+   ```sh
+   aws apigateway update-stage --rest-api-id <rest-api-id> --stage-name <stage-name> --patch-operations op=replace,path=/tracingEnabled,value=true
+   ```
+
+2. **Verify X-Ray Tracing is Enabled:**
+   After enabling X-Ray tracing, verify that it is correctly set up by describing the stage.
+
+   ```sh
+   aws apigateway get-stage --rest-api-id <rest-api-id> --stage-name <stage-name>
+   ```
+
+   Look for the `tracingEnabled` field in the output to ensure it is set to `true`.
+
+3. **Enable X-Ray Tracing for All Stages (Optional):**
+   If you want to ensure that X-Ray tracing is enabled for all stages of a specific API Gateway, you can loop through all stages and enable tracing.
+
+   ```sh
+   stages=$(aws apigateway get-stages --rest-api-id <rest-api-id> --query 'item[*].stageName' --output text)
+   for stage in $stages; do
+       aws apigateway update-stage --rest-api-id <rest-api-id> --stage-name $stage --patch-operations op=replace,path=/tracingEnabled,value=true
+   done
+   ```
+
+4. **Automate the Process with a Script (Optional):**
+   To ensure that X-Ray tracing is always enabled, you can create a script that checks and enables X-Ray tracing for all stages periodically.
+
+   ```sh
+   #!/bin/bash
+   rest_api_id=<rest-api-id>
+   stages=$(aws apigateway get-stages --rest-api-id $rest_api_id --query 'item[*].stageName' --output text)
+   for stage in $stages; do
+       tracing_enabled=$(aws apigateway get-stage --rest-api-id $rest_api_id --stage-name $stage --query 'tracingEnabled' --output text)
+       if [ "$tracing_enabled" != "true" ]; then
+           aws apigateway update-stage --rest-api-id $rest_api_id --stage-name $stage --patch-operations op=replace,path=/tracingEnabled,value=true
+       fi
+   done
+   ```
+
+By following these steps, you can ensure that X-Ray tracing is enabled for your API Gateway stages, helping you to prevent the misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of API Gateway X-Ray not being enabled in AWS API Gateway using Python scripts, you can follow these steps:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed, which is the AWS SDK for Python. You can install it using pip if you haven't already.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+   ```bash
+   aws configure
+   ```
+
+   Alternatively, you can set the environment variables directly in your script:
+
+   ```python
+   import os
+
+   os.environ['AWS_ACCESS_KEY_ID'] = 'your_access_key_id'
+   os.environ['AWS_SECRET_ACCESS_KEY'] = 'your_secret_access_key'
+   os.environ['AWS_DEFAULT_REGION'] = 'your_default_region'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>